dataplane/globalbase.cpp

```diff
@@ -1612,6 +1613,7 @@ balancer_real_id_t* generation::rebuild_service_ring_one_wrr(
         const balancer_real_id_t* const do_not_exceed,
         const balancer_service_t* service)
 {
+	utils::Deferer defer([]() { YADECAP_MEMORY_BARRIER_COMPILE; });
 	for (uint32_t real_idx = service->real_start;
```

Comment:
IMHO this makes code unnecessary complex. Maybe with more complex logic it will be fine, but here I feel like just doing
```cpp
//...
			if (start == do_not_exceed)
			{
				YANET_LOG_ERROR("Balancer service exceeded ring chunk bounds\n");
				YADECAP_MEMORY_BARRIER_COMPILE
				return start;
			}
//...
	YADECAP_MEMORY_BARRIER_COMPILE;
	return start;
}
```
is sufficient enough

Reply:
We need to `YADECAP_MEMORY_BARRIER_COMPILE` at each function exit path. Using it at each existing function exit doesn't convey this intent to future function changes.

How about using a `DEFERED_MEMORY_BARRIER_COMPILE` macro?

Reply:
Yeah, maybe your solution makes more sense that I've thought initially.
smth like this will suffice?
```cpp
struct MemoryBarrierGuard {
    ~MemoryBarrierGuard() {
        YADECAP_MEMORY_BARRIER_COMPILE;
    }
};
```

And do that at the start of every method:
```cpp
Result cControlPlane::init(bool use_kernel_interface)
{
    MemoryBarrierGuard guard;
    //...
```